enwaytech/darknet_ros
# [darknet_ros] set g++ as CUDA compiler, add message regarding CUDA usage

## Additions & Changes
This PR set the CUDA compiler flag to g++, since darknet cannot be built using ccache. This overrides the `catkin build` compiler and always uses g++ regardless of the `CC` and `CXX` flags.

ref: https://stackoverflow.com/a/38532883

## How to test & expected outcomes
> **Note to the reviewer:** The instructions below might not be complete. If you think more tests are necessary, please do not hesitate to perform them and to record the result in a comment in this PR.

1. If you don't have CUDA installed, the build will be without CUDA support. you should get the warning `NO CUDA DETECTED. Building without CUDA`. Unfortunately, warning have the same white color as normal prints, so it is a bit hard to notice in between all the make and catkin prints.
2. If you have CUDA installed, try building with and without this PR change. Without it, the build should fail with error `/usr/bin/ccache: invalid option -- 'E'`

## Recommended reviewer
Linh

### Checklist (tick off when done or remove when not relevant):
- [x] Set labels (do not forget a review time estimation label)

[sc-12509]
